### ISSUE TITLE
Exception Value:	 Model instances passed to related filters must be saved.

### ISSUE THREAD

**Issue by LekiaAnonim** [wagtail/wagtail#11405]
When building a form in wagtail==5.2.2, an Exception Value Error 'Model instances passed to related filters must be saved was raised. 
Django version ==5.0

### Issue Summary
My Model is as follows:
```python
class ContactFormField(AbstractFormField):
    page = ParentalKey('ContactFormPage', on_delete=models.CASCADE, related_name='contat_form_fields')

class ContactFormPage(AbstractEmailForm):
    template = 'home/connect.html'
    intro = RichTextField(blank=True)
    thank_you_text = RichTextField(blank=True)
    content_panels = AbstractEmailForm.content_panels + [
        FormSubmissionsPanel(),
        FieldPanel('intro'),
        InlinePanel('contat_form_fields', label="Form fields"),
        FieldPanel('thank_you_text'),
        MultiFieldPanel([
            FieldRowPanel([
                FieldPanel('from_address', classname="col6"),
                FieldPanel('to_address', classname="col6"),
            ]),
            FieldPanel('subject'),
        ], "Email"),
    ]

    def serve(self, request, *args, **kwargs):
        if request.method == 'POST':
            form = self.get_form(request.POST, page=self, user=request.user)

            if form.is_valid():
                self.process_form_submission(form)
                
                # Update the original landing page context with other data
                landing_page_context = self.get_context(request)
                landing_page_context['email'] = form.cleaned_data['email']

                return render(
                    request,
                    self.get_landing_page_template(request),
                    landing_page_context
                )
        else:
            form = self.get_form(page=self, user=request.user)

        context = self.get_context(request)
        context['form'] = form
        return render(
            request,
            self.get_template(request),
            context
        )
```

### Steps to Reproduce

1. (for example) Start a new project with `wagtail start myproject`
2. Create a form in Model following the steps at https://docs.wagtail.org/en/stable/reference/contrib/forms/index.html.
3. Navigate to Wagtail admin and create a new form page.

### Technical details

-   Python version: Run `python --version`.
-   Django version: Look in your requirements.txt, or run `pip show django | grep Version`.
-   Wagtail version: Look at the bottom of the Settings menu in the Wagtail admin, or run `pip show wagtail | grep Version:`.
-   Browser version: You can use https://www.whatsmybrowser.org/ to find this out.

**Comment by gasman** [comment]
Thanks for the report @LekiaAnonim! I can confirm that I'm seeing the same issue when running on Django 5, with any form using `FormSubmissionsPanel()` (the `serve` method is not necessary here). Tested with a fresh Wagtail instance with the following in `home/models.py`:

```python
from django.db import models

from wagtail.fields import RichTextField
from wagtail.models import Page
from wagtail.contrib.forms.models import AbstractFormField, AbstractEmailForm
from modelcluster.fields import ParentalKey
from wagtail.admin.panels import FieldPanel, InlinePanel, MultiFieldPanel, FieldRowPanel
from wagtail.contrib.forms.panels import FormSubmissionsPanel


class HomePage(Page):
    pass


class ContactFormField(AbstractFormField):
    page = ParentalKey('ContactFormPage', on_delete=models.CASCADE, related_name='contat_form_fields')

class ContactFormPage(AbstractEmailForm):
    intro = RichTextField(blank=True)
    thank_you_text = RichTextField(blank=True)
    content_panels = AbstractEmailForm.content_panels + [
        FormSubmissionsPanel(),
        FieldPanel('intro'),
        InlinePanel('contat_form_fields', label="Form fields"),
        FieldPanel('thank_you_text'),
        MultiFieldPanel([
            FieldRowPanel([
                FieldPanel('from_address', classname="col6"),
                FieldPanel('to_address', classname="col6"),
            ]),
            FieldPanel('subject'),
        ], "Email"),
    ]
```

**Comment by vcatlett** [comment]
The changes to `wagtail/contrib/forms/panels.py` by @gasman in [PR 11406](https://github.com/wagtail/wagtail/pull/11406) solve this issue for me.

**Comment by thibaudcolas** [comment]
👋 fixed and backported for release in Wagtail 5.2.3 (no date yet)
